tests/unit/backend/recipe/algorithm/test_SmoothDataExcludingPeaksAlgo.py

```diff
@@ -0,0 +1,70 @@
+import unittest.mock as mock
+
+import pytest
+
+with mock.patch.dict(
+    "sys.modules",
+    {
+        "snapred.backend.log": mock.Mock(),
+        "snapred.backend.log.logger": mock.Mock(),
+    },
+):
+    from mantid.simpleapi import (
+        DeleteWorkspace,
+        LoadNexusProcessed,
+        mtd,
+    )
+    from snapred.backend.dao.calibration.Calibration import Calibration
+    from snapred.backend.dao.CrystallographicInfo import CrystallographicInfo
+    from snapred.backend.dao.SmoothDataExcludingPeaksIngredients import SmoothDataExcludingPeaksIngredients
+    from snapred.backend.recipe.algorithm.SmoothDataExcludingPeaksAlgo import SmoothDataExcludingPeaks
+    from snapred.meta.Config import Resource
+
+    def setup():
+        pass
+
+    def teardown():
+        workspaces = mtd.getObjectNames()
+
+        for workspace in workspaces:
+            try:
+                DeleteWorkspace(workspace)
+            except ValueError:
+                print(f"Workspace {workspace} doesn't exist!")
+
+    @pytest.fixture(autouse=True)
+    def _setup_teardown():
+        setup()
+        yield
+        teardown()
+
+    def test_SmoothDataExcludingPeaksAlgo():
+        # input data
+        testWorkspaceFile = "inputs/strip_peaks/DSP_58882_cal_CC_Column_spectra.nxs"
+
+        # loading test workspace
+        test_ws_name = "test_ws"
+        LoadNexusProcessed(Filename=Resource.getPath(testWorkspaceFile), OutputWorkspace=test_ws_name)
+
+        # load crystal info for testing
+        crystalInfo = CrystallographicInfo.parse_raw(Resource.read("inputs/purge_peaks/input_crystalInfo.json"))
+
+        # load instrument state for testing
+        instrumentState = Calibration.parse_raw(
+            Resource.read("inputs/purge_peaks/input_parameters.json")
+        ).instrumentState
+
+        # populate ingredients
+        smoothDataIngredients = SmoothDataExcludingPeaksIngredients(
+            CrystalInfo=crystalInfo, InstrumentState=instrumentState
+        )
+
+        # initialize and run smoothdata algo
+        smoothDataAlgo = SmoothDataExcludingPeaks()
+        smoothDataAlgo.initialize()
+        smoothDataAlgo.setProperty("InputWorkspace", test_ws_name)
+        smoothDataAlgo.setProperty("SmoothDataExcludingPeaksIngredients", smoothDataIngredients.json())
+        smoothDataAlgo.execute()
+
+        assert smoothDataAlgo.getProperty("InputWorkspace").value == "test_ws"
+        assert smoothDataAlgo.getProperty("OutputWorkspace").value == "SmoothedDataExcludingPeaks"
```

Comment:
This test really only checks that `execute` runs without errors, and that the two properties are set to string values.  Are there plans for tests of behavior?

Reply:
Robert's algo that gets called in mine calculates the weights and also produces predicts the peaks. The logic in my algo is basic and contains the results of the weight calculation. So basically all I have to do is concatenate the original y values with the weight array to remove the peaks and return those new y values as the x values stay the same. I did some tests in printing the array values after they are concatenated and it seems the return the right values as far as multiple areas within the array having zero values. What else could I test here other than looking for the input and output workspaces?

Reply:
As just discussed, after status (8/2/2023) one idea is to calculate the weighted chi2 treating the output spline as a model and to check this is less than a reasonable value (say ~3). A second idea is to manually generate a "correct" spline fit and then take the ratio of it's integral and the integral of "input data minus spline". A test could then check if this ratio is matched (within some tolerance).